runtime/src/relaxed/cpp/MemoryImpl.cpp

```diff
@@ -3,7 +3,7 @@
  * that can be found in the LICENSE file.
  */
 #include "Memory.h"
-#include "MemoryPrivate.hpp"
+#include "../../legacy_memory_manager/cpp/MemoryPrivate.hpp" // Fine, because this module is a part of legacy MM.
```

Comment:
Can we move `strict` and `relaxed` into `legacy_memory_manager` later?

Reply:
Yes, we can make `legacy_memory_manager` depend upon a flag `USE_STRICT_MEMORY_MODEL` and compile it twice with 2 different values.

Reply:
I actually meant just moving the directories, not merging the modules.

Reply:
We will need to redo our c++ build plugin for that, but it's possible.